### ISSUE TITLE
2.17 -> 3.0 docker upgrade fails with postgres index error

### ISSUE THREAD

**Issue by glensc** [WeblateOrg/weblate#2118]
https://docs.weblate.org/en/weblate-3.0/admin/deployments.html#upgrading-docker-container

tried to upgrade 2.17.1-1 -> 3.0.1-4
```diff
 services:
   weblate:
-    image: weblate/weblate:2.17.1-1
+    image: weblate/weblate:3.0.1-4
```

```
# docker-compose logs database
Attaching to weblate_database_1
database_1  | LOG:  database system was shut down at 2018-07-10 12:27:53 UTC
database_1  | LOG:  MultiXact member wraparound protections are now enabled
database_1  | LOG:  database system is ready to accept connections
database_1  | LOG:  autovacuum launcher started
database_1  | ERROR:  relation "lang_plural" does not exist at character 157
database_1  | STATEMENT:  SELECT "lang_plural"."id", "lang_plural"."source", "lang_plural"."number", "lang_plural"."equation", "lang_plural"."type", "lang_plural"."language_id" FROM "lang_plural" WHERE ("lang_plural"."language_id" = 254 AND "lang_plural"."language_id" = 254 AND "lang_plural"."source" = 0)
database_1  | LOG:  database system was interrupted; last known up at 2018-07-10 12:28:20 UTC
database_1  | LOG:  database system was not properly shut down; automatic recovery in progress
database_1  | LOG:  redo starts at 0/1110D4E0
database_1  | LOG:  record with zero length at 0/11145B08
database_1  | LOG:  redo done at 0/11145AD8
database_1  | LOG:  last completed transaction was at log time 2018-07-10 12:28:31.180646+00
database_1  | LOG:  MultiXact member wraparound protections are now enabled
database_1  | LOG:  database system is ready to accept connections
database_1  | LOG:  autovacuum launcher started
```

restarting `database` container does not help

**Comment by glensc** [comment]
seems this resolved by itself. i downed all services (to move volumes around) and when i started again. migrate finished successfully.

**Comment by glensc** [comment]
i was able to reproduce the same problem on second node:
1. take posgresql dump from 2.17
2. extract it for posgresql data
3. start containers
4. error
5. stop containers
6. start containers
7. success

so, for some reason it fails first time and succeeds on second run.

**Comment by nijel** [comment]
Did you get some error on the weblate container? If so, posting it would be more useful than the failed SQL query...

**Comment by nijel** [comment]
Okay, I've managed to reproduce it now:

```
weblate_1   | Traceback (most recent call last):
weblate_1   |   File "/usr/local/bin/weblate", line 11, in <module>
weblate_1   |     sys.exit(main())
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/weblate/runner.py", line 33, in main
weblate_1   |     execute_from_command_line(argv)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 371, in execute_from_command_line
weblate_1   |     utility.execute()
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 365, in execute
weblate_1   |     self.fetch_command(subcommand).run_from_argv(self.argv)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 288, in run_from_argv
weblate_1   |     self.execute(*args, **cmd_options)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 335, in execute
weblate_1   |     output = self.handle(*args, **options)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/core/management/commands/migrate.py", line 223, in handle
weblate_1   |     self.verbosity, self.interactive, connection.alias, apps=post_migrate_apps, plan=plan,
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/core/management/sql.py", line 51, in emit_post_migrate_signal
weblate_1   |     **kwargs
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/dispatch/dispatcher.py", line 178, in send
weblate_1   |     for receiver in self._live_receivers(sender)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/dispatch/dispatcher.py", line 178, in <listcomp>
weblate_1   |     for receiver in self._live_receivers(sender)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/weblate/lang/models.py", line 340, in setup_lang
weblate_1   |     Language.objects.setup(False)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/manager.py", line 82, in manager_method
weblate_1   |     return getattr(self.get_queryset(), name)(*args, **kwargs)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/weblate/lang/models.py", line 293, in setup
weblate_1   |     defaults=plural_data,
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/fields/related_descriptors.py", line 629, in get_or_create
weblate_1   |     return super(RelatedManager, self.db_manager(db)).get_or_create(**kwargs)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/manager.py", line 82, in manager_method
weblate_1   |     return getattr(self.get_queryset(), name)(*args, **kwargs)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 487, in get_or_create
weblate_1   |     return self.get(**lookup), False
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 397, in get
weblate_1   |     num = len(clone)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 254, in __len__
weblate_1   |     self._fetch_all()
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 1179, in _fetch_all
weblate_1   |     self._result_cache = list(self._iterable_class(self))
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/query.py", line 53, in __iter__
weblate_1   |     results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/models/sql/compiler.py", line 1068, in execute_sql
weblate_1   |     cursor.execute(sql, params)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 100, in execute
weblate_1   |     return super().execute(sql, params)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 68, in execute
weblate_1   |     return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 77, in _execute_with_wrappers
weblate_1   |     return executor(sql, params, many, context)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 85, in _execute
weblate_1   |     return self.cursor.execute(sql, params)
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/utils.py", line 89, in __exit__
weblate_1   |     raise dj_exc_value.with_traceback(traceback) from exc_value
weblate_1   |   File "/usr/local/lib/python3.5/dist-packages/django/db/backends/utils.py", line 85, in _execute
weblate_1   |     return self.cursor.execute(sql, params)
weblate_1   | django.db.utils.ProgrammingError: relation "lang_plural" does not exist
weblate_1   | LINE 1: ..._plural"."type", "lang_plural"."language_id" FROM "lang_plur...
weblate_1   |                                                              ^
```

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed in docker image 3.0.1-6.

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
